schematic/src/x_widgets.c

```diff
@@ -114,3 +262,57 @@ x_widgets_show_in_dock (GtkWidget* wbook, GtkWidget* widget)
     gtk_widget_set_visible (wbook, TRUE);
   }
 }
+
+
+
+/*! \brief Shows a widget as a dialog box
+ *
+ *  \param [in]     w_current The toplevel environment.
+ *  \param [in]     widget Widget to show
+ *  \param [in,out] dialog Dialog which will be a parent of the widget
+ *  \param [in]     title  Dialog's title
+ *  \param [in]     ini_group Config file section for dialog geometry
+ */
+static void
+x_widgets_show_in_dialog (GschemToplevel* w_current,
+                          GtkWidget*      widget,
+                          GtkWidget**     dialog,
+                          const gchar*    title,
+                          const gchar*    ini_group)
+{
+  g_return_if_fail (widget != NULL);
+
+  if (*dialog != NULL)
+  {
+    gtk_window_present (GTK_WINDOW (*dialog));
+    return;
+  }
+
+  GtkWidget* dlg = gschem_dialog_new_with_buttons(
+    title,
+    GTK_WINDOW (w_current->main_window),
+    0,
```

Comment:
When compiling using `g++`, it reports a missing cast to *GtkDialogFlags* here. Please fix this.